kernel/h1_syscalls/src/reset.rs

```diff
@@ -0,0 +1,70 @@
+// Copyright 2021 lowRISC contributors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+use h1::hil::reset::Reset;
+use kernel::{AppId, AppSlice, Callback, Driver, ReturnCode, Shared};
+
+pub const DRIVER_NUM: usize = 0x40070;
+
+pub struct ResetSyscall<'a> {
+    reset: &'a dyn Reset,
+}
+
+impl<'a> ResetSyscall<'a> {
+    pub fn new(reset: &'a dyn Reset) -> ResetSyscall<'a> {
+        ResetSyscall {
+            reset: reset,
+        }
+    }
+
+    fn reset(&self) -> ReturnCode {
+        self.reset.reset();
+
+        // This should never return. If it did, something went wrong.
```

Comment:
Is there no such thing as `UNREACHABLE()` to use here?

Reply:
I updated the comment. The compiler would flag it if `self.reset.reset_chip()` changed its return type and actually returned.

Reply:
:+1: